pkg/terraform/exec/plugins/vsphereprivate/resource_vsphereprivate_import_ova.go

```diff
@@ -183,11 +183,17 @@ func findImportOvaParams(client *vim25.Client, datacenter, cluster, datastore, n
 	// available for use on the HostSystem we will import the
 	// OVA to.
 	for _, hostObj := range hosts {
-		hostObj.Properties(ctx, hostObj.Reference(), []string{"network", "datastore"}, &hostSystemManagedObject)
-
+		foundDatastore = false
+		foundNetwork = false
```

Comment:
Nice catch here.

I think there needs to be some work on the error messages on lines 214-219. Let's say that there are two hosts. The first host contains the datastore but not the network. The second host is in maintenance mode. At the end of the for loop, both `foundDatastore` and `foundNetwork` will be false. The error returned will indicate that a host with the datastore was not found, which is not true. I think we can combine the two errors into something like the following.

"failed to find a host in the cluster that contains the provided datacenter and network"

If we are not distinguishing in the error message between missing the datacenter or missing the network, then we can keep the `foundDatastore` and `foundNetwork` in the scope of the for loop. We can just return from the function when the host is found.

Reply:
This is a good point. I had not really considered the logic of the error messages themselves. Resolved with https://github.com/openshift/installer/commit/ceedd7da83a3b2b99d2ece59723871ef5d5656e9